ouroboros-network/src/Ouroboros/Network/PeerSelection/PeerStateActions.hs

```diff
@@ -854,7 +854,7 @@ withPeerStateActions PeerStateActionsArguments {
     monitorPeerConnection PeerConnectionHandle { pchPeerStatus, pchAppHandles } =
          p  <$> readTVar pchPeerStatus
             <*> (g <$> traverse f pchAppHandles)
-        `orElse` throwSTM MonitorPeerConnectionBlocked
+        `orElse` throwSTM MonitorPeerConnectionBlocked -- ^ looks like throwSTM here is dead code?
```

Comment:
```suggestion
        `orElse` throwSTM MonitorPeerConnectionBlocked
```
it is, and if we introduce a bug in `monitorPeerConnection` we'll discover it in a failing test.